Library/Formula/ice.rb

```diff
@@ -42,17 +30,35 @@ def install
 
       # BerkeleyDB dislikes parallel builds
       ENV.deparallelize
-
-      # --enable-compat185 is necessary because our build shadows
-      # the system berkeley db 1.x
       args = %W[
         --disable-debug
         --prefix=#{libexec}
         --mandir=#{libexec}/man
         --enable-cxx
-        --enable-compat185
       ]
 
+      if build.with? "java"
+        java_home = ENV["JAVA_HOME"] = `/usr/libexec/java_home -v 1.7`.chomp
```

Comment:
Yes all of this settings including `ENV.append("CPPFLAGS", ...)` are already handed by `:java`. `args << "--enable-java"` should be enough.